daffodil-core/src/main/scala/org/apache/daffodil/dpath/NodeInfoUtils.scala

```diff
@@ -113,24 +113,34 @@ object NodeInfoUtils {
       case (Float, _) => (Float, Float)
       case (_, Integer) => (Integer, Integer)
       case (Integer, _) => (Integer, Integer)
+      case (lat, NonNegativeInteger) if (lat.isInstanceOf[SignedInteger.Kind]) => (Integer, Integer)
```

Comment:
Is there a way to remove the asInstanceOf stuff? I think you might need to use parenthesis to match the instance instead of the companion object?
```scala
case (SignedInteger(), NonNegativeInteger) => ...
```
Or maybe you needs to be bound to a variable?
```scala
case (_: SignedInteger, NonNegativeInteger) => ...
```
I need to reference myself of scala pattern matching.

Reply:
The second case works and cleans up the code a lot.  I think I was overthinking things with the if statements.
I've pushed up the changes.